File: ticketProject7-CrossPlatform/TS-crossp/protobuf3/ProtoUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using services;
using model;
using Chat.Protocol;
using proto = Chat.Protocol;

namespace protobuf
{
    static class ProtoUtils
    {
        public static TSRequest createLoginRequest(model.Seller user)
        {
            proto.Seller userDTO = new proto.Seller{ Id = user.Id.ToString() , FirstName = user.FirstName, LastName = user.LastName, Username = user.Username, Password = user.Password };
            TSRequest request = new TSRequest { Type = TSRequest.Types.Type.Login, Seller = userDTO };
            return request;
        }

        public static TSRequest createLogoutRequest(model.Seller user)
        {
            proto.Seller userDTO = new proto.Seller { Id = user.Id.ToString(), FirstName = user.FirstName, LastName = user.LastName, Username = user.Username, Password = user.Password };
            TSRequest request = new TSRequest { Type = TSRequest.Types.Type.Logout, Seller = userDTO };
            return request;
        }

        public static TSRequest createAddBuyerRequest(model.Buyer buyer)
        {
            proto.Buyer buyerDTO = new proto.Buyer { Id = buyer.Id.ToString(), FirstName = buyer.FirstName, LastName = buyer.LastName, IdAR = buyer.IdAR.ToString(), NoTickets = buyer.NoTickets.ToString() }; 
            TSRequest request = new TSRequest { Type = TSRequest.Types.Type.AddBuyer, Buyer = buyerDTO };
            return request;
        }

        public static TSRequest createBuyTicketRequest(model.ArtisticRepresentation artisticRepresentation)
        {
            proto.ArtisticRepresentation arDTO = new proto.ArtisticRepresentation { Id = artisticRepresentation.Id.ToString(), ArtistName = artisticRepresentation.ArtistName, Data = artisticRepresentation.Data.ToString(), Location = artisticRepresentation.Location, AvailableSeats = artisticRepresentation.AvailableSeats.ToString(), SoldSeats = artisticRepresentation.SoldSeats.ToString() }; 
            TSRequest request = new TSRequest { Type = TSRequest.Types.Type.BuyTicket, ArtisticRepresentation = arDTO };
            return request;
        }

        public static TSRequest createFindSellerByUsernameRequest(model.Seller seller)
        {
            proto.Seller userDTO = new proto.Seller { Id = seller.Id.ToString(), FirstName = seller.FirstName, LastName = seller.LastName, Username = seller.Username, Password = seller.Password }; 
            TSRequest request = new TSRequest { Type = TSRequest.Types.Type.FindSellerByUsername, Seller = userDTO };
            return request;
        }

        public static TSRequest createGetAllARRequest()
        {
            proto.TSRequest request = new proto.TSRequest { Type = proto.TSRequest.Types.Type.GetAllAr };
            return request;
        }

        public static TSRequest createFGetAllARFRomDateRequest(DateTime data)
        {
            proto.ArtisticRepresentation arDTO = new proto.ArtisticRepresentation { Id = "1", ArtistName = "sdd", Data = data.ToString(), Location = "sazdf", AvailableSeats = "1", SoldSeats = "1" };
            proto.TSRequest request = new proto.TSRequest { Type = proto.TSRequest.Types.Type.GetAllArfromDate, ArtisticRepresentation = arDTO };
            return request;
        }

        //-------------------------
        public static TSResponse createOkResponse()
        {
            TSResponse response = new TSResponse { Type = TSResponse.Types.Type.Ok };
            return response;
        }


        public static TSResponse createErrorResponse(String text)
        {
            TSResponse response = new TSResponse
            {
                Type = TSResponse.Types.Type.Error,
                Error = text
            };
            return response;
        }

        public static TSResponse createLoginResponse(model.Seller seller)
        {
            proto.Seller sellerDTO = new proto.Seller { Id = seller.Id.ToString(), FirstName = seller.FirstName, LastName = seller.LastName, Username = seller.Username, Password = seller.Password };
            proto.TSResponse response = new proto.TSResponse { Type = TSResponse.Types.Type.Login, Seller = sellerDTO };
            return response;
        }

        public static TSResponse createLogoutResponse()
        {
           TSResponse response = new TSResponse { Type = TSResponse.Types.Type.Logout };
            return response;
        }

        public static TSResponse createBoughtTicketResponse(model.ArtisticRepresentation artisticRepresentation)
        {
            TSResponse response = new TSResponse { Type = TSResponse.Types.Type.BoughtTicket };

            proto.ArtisticRepresentation arDTO = new proto.ArtisticRepresentation { Id = artisticRepresentation.Id.ToString(), ArtistName = artisticRepresentation.ArtistName, Data = artisticRepresentation.Data.ToString(), Location = artisticRepresentation.Location, AvailableSeats = artisticRepresentation.AvailableSeats.ToString(), SoldSeats = artisticRepresentation.SoldSeats.ToString() };
            response.ArtisticRepresentation.Add(arDTO);

            return response;
        }

        public static TSResponse createAddBuyerResponse(model.Buyer buyer)
        {
            if (buyer == null)
            {
                TSResponse response = new TSResponse { Type = TSResponse.Types.Type.Ok };
                return response;
            }
            else
            {
                proto.Buyer buyerDTO = new proto.Buyer { Id = buyer.Id.ToString(), FirstName = buyer.FirstName, LastName = buyer.LastName, IdAR = buyer.IdAR.ToString(), NoTickets = buyer.NoTickets.ToString() };
                TSResponse response = new TSResponse { Type = TSResponse.Types.Type.AddBuyer, Buyer = buyerDTO };
                return response;
            }
        }

        public static TSResponse createBuyTicketResponse(model.ArtisticRepresentation artisticRepresentation)
        {
            if (artisticRepresentation == null)
            {
                proto.ArtisticRepresentation arDTO = null;
                TSResponse response = new TSResponse { Type = TSResponse.Types.Type.Ok };
                return response;
            }
            else
            {
                proto.ArtisticRepresentation arDTO = new proto.ArtisticRepresentation { Id = artisticRepresentation.Id.ToString(), ArtistName = artisticRepresentation.ArtistName, Data = artisticRepresentation.Data.ToString(), Location = artisticRepresentation.Location, AvailableSeats = artisticRepresentation.AvailableSeats.ToString(), SoldSeats = artisticRepresentation.SoldSeats.ToString() };

                //TSResponse response = new TSResponse { Type = TSResponse.Types.Type.BuyTicket, ArtisticRepresentation = arDTO };
                //return response;
                TSResponse response = new TSResponse { Type = TSResponse.Types.Type.BuyTicket};
                response.ArtisticRepresentation.Add(arDTO);
                return response;
            }
        }

        public static TSResponse createGetAllARResponse(IEnumerable<model.ArtisticRepresentation> ars)
        {
            TSResponse response = new TSResponse { Type = TSResponse.Types.Type.GetAllAr };

            foreach (model.ArtisticRepresentation artisticRepresentation in ars)
            {
                proto.ArtisticRepresentation arDTO = new proto.ArtisticRepresentation { Id = artisticRepresentation.Id.ToString(), ArtistName = artisticRepresentation.ArtistName, Data = artisticRepresentation.Data.ToString(), Location = artisticRepresentation.Location, AvailableSeats = artisticRepresentation.AvailableSeats.ToString(), SoldSeats = artisticRepresentation.SoldSeats.ToString() };
                //proto.ArtisticRepresentation arDTO = new proto.ArtisticRepresentation { Id = artisticRepresentation.Id.ToString(), ArtistName = artisticRepresentation.ArtistName, Data = getStringFromDateTime(artisticRepresentation.Data), Location = artisticRepresentation.Location, AvailableSeats = artisticRepresentation.AvailableSeats.ToString(), SoldSeats = artisticRepresentation.SoldSeats.ToString() };
                response.ArtisticRepresentation.Add(arDTO);
            }

            return response;
        }

        public static String getStringFromDateTime(DateTime data)
        {
            String result = "";
            result = result + data.Year + "-" + data.Month + "-" + data.Day + "T" + data.Hour + ":" + data.Minute + ":" + data.Second;
            return result;
        }

        public static TSResponse createGetAllARFromDateResponse(IEnumerable<model.ArtisticRepresentation> ars)
        {
            TSResponse response = new TSResponse { Type = TSResponse.Types.Type.GetAllArfromDate };

            foreach (model.ArtisticRepresentation artisticRepresentation in ars)
            {
                proto.ArtisticRepresentation arDTO = new proto.ArtisticRepresentation { Id = artisticRepresentation.Id.ToString(), ArtistName = artisticRepresentation.ArtistName, Data = artisticRepresentation.Data.ToString(), Location = artisticRepresentation.Location, AvailableSeats = artisticRepresentation.AvailableSeats.ToString(), SoldSeats = artisticRepresentation.SoldSeats.ToString() };
                //proto.ArtisticRepresentation arDTO = new proto.ArtisticRepresentation { Id = artisticRepresentation.Id.ToString(), ArtistName = artisticRepresentation.ArtistName, Data = getStringFromDateTime(artisticRepresentation.Data), Location = artisticRepresentation.Location, AvailableSeats = artisticRepresentation.AvailableSeats.ToString(), SoldSeats = artisticRepresentation.SoldSeats.ToString() };
                response.ArtisticRepresentation.Add(arDTO);
            }

            return response;
        }

        public static TSResponse createFindSellerByUsernameResponse(model.Seller seller)
        {
            proto.Seller sellerDTO = new proto.Seller{ Id = seller.Id.ToString(), FirstName = seller.FirstName, LastName = seller.LastName, Username = seller.Username, Password = seller.Password };
            TSResponse response = new TSResponse { Type = TSResponse.Types.Type.FindSellerByUsername, Seller = sellerDTO };
            return response;
        }
        //----------------------------

        public static String getError(TSResponse response)
        {
            String errorMessage = response.Error;
            return errorMessage;
        }

        public static model.Seller getSeller(TSResponse response)
        {
            model.Seller seller = new model.Seller()
            {
                Id = long.Parse(response.Seller.Id),
                FirstName = response.Seller.FirstName,
                LastName = response.Seller.LastName,
                Username = response.Seller.Username,
                Password = response.Seller.Password
            };
            return seller;
        }

        public static IEnumerable<model.ArtisticRepresentation> getArtisticRepresentations(TSResponse response)
        {
            List<model.ArtisticRepresentation> ars = new List<model.ArtisticRepresentation>();
            for (int i = 0; i < response.ArtisticRepresentation.Count; i++)
            {
                model.ArtisticRepresentation ar = new model.ArtisticRepresentation("1L", DateTime.Now, "sdc", 2L, 2L);
                ar.Id = long.Parse(response.ArtisticRepresentation[i].Id);
                ar.ArtistName = response.ArtisticRepresentation[i].ArtistName;
                ar.Data = DateTime.Parse(response.ArtisticRepresentation[i].Data);
                ar.Location = response.ArtisticRepresentation[i].Location;
                ar.AvailableSeats = long.Parse(response.ArtisticRepresentation[i].AvailableSeats);
                ar.SoldSeats = long.Parse(response.ArtisticRepresentation[i].SoldSeats);

                ars.Add(ar);
            }
            return ars;
        }

        public static model.Seller getSeller(TSRequest request)
        {
            model.Seller seller = new model.Seller();
            seller.Id = long.Parse(request.Seller.Id);
            seller.FirstName = request.Seller.FirstName;
            seller.LastName = request.Seller.LastName;
            seller.Username = request.Seller.Username;
            seller.Password = request.Seller.Password;
            return seller;
        }

        public static model.Buyer getBuyer(TSRequest request)
        {
            model.Buyer buyer = new model.Buyer(1L, "scd", "sdc", 2L);
            buyer.Id = long.Parse(request.Buyer.Id);
            buyer.IdAR = long.Parse(request.Buyer.IdAR);
            buyer.FirstName = request.Buyer.FirstName;
            buyer.LastName = request.Buyer.LastName;
            buyer.NoTickets = long.Parse(request.Buyer.NoTickets);
            return buyer;
        }

        public static model.ArtisticRepresentation getArtisticRepresentation(TSRequest request)
        {
            model.ArtisticRepresentation ar = new model.ArtisticRepresentation("1", DateTime.Now, "sdc", 2L, 2L);
            ar.Id = long.Parse(request.ArtisticRepresentation.Id);
            ar.ArtistName = request.ArtisticRepresentation.ArtistName;
            ar.Data = DateTime.Parse(request.ArtisticRepresentation.Data);
            ar.Location = request.ArtisticRepresentation.Location;
            ar.AvailableSeats = long.Parse(request.ArtisticRepresentation.AvailableSeats);
            ar.SoldSeats = long.Parse(request.ArtisticRepresentation.SoldSeats);
            return ar;
        }

    }
}

File: ticketProject7-CrossPlatform/TS-cross/TicketNetworking/src/main/java/ticket/network/protobuffprotocol/ProtoUtils.java
package ticket.network.protobuffprotocol;

import ticket.model.ArtisticRepresentation;
import ticket.model.Buyer;
import ticket.model.Seller;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class ProtoUtils {
    public static TSProtobufs.TSRequest createLoginRequest(Seller user){
        TSProtobufs.Seller userDTO = TSProtobufs.Seller.newBuilder().setId(user.getId().toString()).setFirstName(user.getFirstName()).setLastName(user.getLastName()).setUsername(user.getUsername()).setPassword(user.getPassword()).build();
        TSProtobufs.TSRequest request= TSProtobufs.TSRequest.newBuilder().setType(TSProtobufs.TSRequest.Type.Login)
                .setSeller(userDTO).build();
        return request;
    }
    public static TSProtobufs.TSRequest createLogoutRequest(Seller user){
        TSProtobufs.Seller userDTO= TSProtobufs.Seller.newBuilder().setId(user.getId().toString()).setFirstName(user.getFirstName()).setLastName(user.getLastName()).setUsername(user.getUsername()).setPassword(user.getPassword()).build();
        TSProtobufs.TSRequest request= TSProtobufs.TSRequest.newBuilder().setType(TSProtobufs.TSRequest.Type.Logout)
                .setSeller(userDTO).build();
        return request;
    }

    public static TSProtobufs.TSRequest createAddBuyerRequest(Buyer buyer){
        TSProtobufs.Buyer buyerDTO = TSProtobufs.Buyer.newBuilder().setId(buyer.getId().toString()).setFirstName(buyer.getFirstName()).setLastName(buyer.getLastName()).setIdAR(buyer.getIdAR().toString()).setNoTickets(buyer.getNoTickets().toString()).build();
        TSProtobufs.TSRequest request= TSProtobufs.TSRequest.newBuilder().setType(TSProtobufs.TSRequest.Type.AddBuyer)
                .setBuyer(buyerDTO).build();
        return request;
    }

    public static TSProtobufs.TSRequest createBuyTicketRequest(ArtisticRepresentation artisticRepresentation){
        TSProtobufs.ArtisticRepresentation arDTO = TSProtobufs.ArtisticRepresentation.newBuilder().setId(artisticRepresentation.getId().toString()).setArtistName(artisticRepresentation.getArtistName()).setData(artisticRepresentation.getData().toString()).setLocation(artisticRepresentation.getLocation()).setAvailableSeats(artisticRepresentation.getAvailableSeats().toString()).setSoldSeats(artisticRepresentation.getSoldSeats().toString()).build();
        TSProtobufs.TSRequest request = TSProtobufs.TSRequest.newBuilder().setType(TSProtobufs.TSRequest.Type.BuyTicket)
                .setArtisticRepresentation(arDTO).build();
        return request;
    }

    public static TSProtobufs.TSRequest createFindSellerByUsernameRequest(Seller seller){
        TSProtobufs.Seller userDTO = TSProtobufs.Seller.newBuilder().setId(seller.getId().toString()).setFirstName(seller.getFirstName()).setLastName(seller.getLastName()).setUsername(seller.getUsername()).setPassword(seller.getPassword()).build();
        TSProtobufs.TSRequest request = TSProtobufs.TSRequest.newBuilder().setType(TSProtobufs.TSRequest.Type.FindSellerByUsername)
                .setSeller(userDTO).build();
        return request;
    }

    public static TSProtobufs.TSRequest createGetAllARRequest(){
        TSProtobufs.TSRequest request = TSProtobufs.TSRequest.newBuilder().setType(TSProtobufs.TSRequest.Type.GetAllAR)
                .build();
        return request;
    }

    public static TSProtobufs.TSRequest createFGetAllARFRomDateRequest(LocalDateTime data){
        TSProtobufs.ArtisticRepresentation arDTO= TSProtobufs.ArtisticRepresentation.newBuilder().setId("1").setArtistName("sss").setData(data.toString()).setLocation("sdvg").setAvailableSeats("1").setSoldSeats("1").build();
        TSProtobufs.TSRequest request = TSProtobufs.TSRequest.newBuilder().setType(TSProtobufs.TSRequest.Type.GetAllARFromDate)
                .setArtisticRepresentation(arDTO).build();
        return request;
    }





    public static TSProtobufs.TSResponse createOkResponse(){
        TSProtobufs.TSResponse response= TSProtobufs.TSResponse.newBuilder()
                .setType(TSProtobufs.TSResponse.Type.Ok).build();
        return response;
    }

    public static TSProtobufs.TSResponse createErrorResponse(String text){
        TSProtobufs.TSResponse response= TSProtobufs.TSResponse.newBuilder()
                .setType(TSProtobufs.TSResponse.Type.Error)
                .setError(text).build();
        return response;
    }

    public static TSProtobufs.TSResponse createLoginResponse(Seller seller){
        TSProtobufs.Seller sellerDTO = TSProtobufs.Seller.newBuilder().setId(seller.getId().toString()).setFirstName(seller.getFirstName()).setLastName(seller.getLastName()).setUsername(seller.getUsername()).setPassword(seller.getPassword()).build();

        TSProtobufs.TSResponse response = TSProtobufs.TSResponse.newBuilder()
                .setType(TSProtobufs.TSResponse.Type.Login)
                .setSeller(sellerDTO).build();
        return response;
    }

    public static TSProtobufs.TSResponse createLogoutResponse(){
        //TSProtobufs.Seller sellerDTO = TSProtobufs.Seller.newBuilder().setId(seller.getId().toString()).setFirstName(seller.getFirstName()).setLastName(seller.getLastName()).setUsername(seller.getUsername()).setPassword(seller.getPassword()).build();

        TSProtobufs.TSResponse response = TSProtobufs.TSResponse.newBuilder()
                .setType(TSProtobufs.TSResponse.Type.Logout)
                .build();
        return response;
    }

    public static TSProtobufs.TSResponse createBoughtTicketResponse(ArtisticRepresentation artisticRepresentation){
//        TSProtobufs.ArtisticRepresentation arDTO = TSProtobufs.ArtisticRepresentation.newBuilder().setId(artisticRepresentation.getId().toString()).setArtistName(artisticRepresentation.getArtistName()).setData(artisticRepresentation.getData().toString()).setLocation(artisticRepresentation.getLocation()).setAvailableSeats(artisticRepresentation.getAvailableSeats().toString()).setSoldSeats(artisticRepresentation.getSoldSeats().toString()).build();
//
//        TSProtobufs.TSResponse response = TSProtobufs.TSResponse.newBuilder()
//                .setType(TSProtobufs.TSResponse.Type.BoughtTicket)
//                .setArtisticRepresentation(0, arDTO).build();
//        return response;
        TSProtobufs.TSResponse.Builder response = TSProtobufs.TSResponse.newBuilder()
                .setType(TSProtobufs.TSResponse.Type.BoughtTicket);


            TSProtobufs.ArtisticRepresentation arDTO = TSProtobufs.ArtisticRepresentation.newBuilder().setId(artisticRepresentation.getId().toString()).setArtistName(artisticRepresentation.getArtistName()).setData(artisticRepresentation.getData().toString()).setLocation(artisticRepresentation.getLocation()).setAvailableSeats(artisticRepresentation.getAvailableSeats().toString()).setSoldSeats(artisticRepresentation.getSoldSeats().toString()).build();
            response.addAllArtisticRepresentation(Collections.singleton(arDTO));


        return response.build();
    }

    public static TSProtobufs.TSResponse createAddBuyerResponse(Buyer buyer){
        if(buyer == null)
        {
            TSProtobufs.TSResponse response = TSProtobufs.TSResponse.newBuilder()
                    .setType(TSProtobufs.TSResponse.Type.Ok)
                    .build();
            return response;
        }
        else {
            TSProtobufs.Buyer buyerDTO = TSProtobufs.Buyer.newBuilder().setId(buyer.getId().toString()).setIdAR(buyer.getIdAR().toString()).setFirstName(buyer.getFirstName()).setLastName(buyer.getLastName()).setNoTickets(buyer.getNoTickets().toString()).build();

            TSProtobufs.TSResponse response = TSProtobufs.TSResponse.newBuilder()
                    .setType(TSProtobufs.TSResponse.Type.AddBuyer)
                    .setBuyer(buyerDTO).build();
            return response;
        }
    }

    public static TSProtobufs.TSResponse createBuyTicketResponse(ArtisticRepresentation artisticRepresentation){
        if(artisticRepresentation == null) {
            TSProtobufs.ArtisticRepresentation arDTO = null;
            TSProtobufs.TSResponse response = TSProtobufs.TSResponse.newBuilder()
                    .setType(TSProtobufs.TSResponse.Type.Ok)
                   .build();
            return response;
        }
        else
        {
            TSProtobufs.ArtisticRepresentation arDTO = TSProtobufs.ArtisticRepresentation.newBuilder().setId(artisticRepresentation.getId().toString()).setArtistName(artisticRepresentation.getArtistName()).setData(artisticRepresentation.getData().toString()).setLocation(artisticRepresentation.getLocation()).setAvailableSeats(artisticRepresentation.getAvailableSeats().toString()).setSoldSeats(artisticRepresentation.getSoldSeats().toString()).build();

            TSProtobufs.TSResponse response = TSProtobufs.TSResponse.newBuilder()
                    .setType(TSProtobufs.TSResponse.Type.BuyTicket)
                    .setArtisticRepresentation(0, arDTO).build();
            return response;
        }
    }

    public static TSProtobufs.TSResponse createGetAllARResponse(Iterable<ArtisticRepresentation> ars){
        TSProtobufs.TSResponse.Builder response = TSProtobufs.TSResponse.newBuilder()
                .setType(TSProtobufs.TSResponse.Type.GetAllAR);

        for(ArtisticRepresentation ar : ars){
            TSProtobufs.ArtisticRepresentation arDTO = TSProtobufs.ArtisticRepresentation.newBuilder().setId(ar.getId().toString()).setArtistName(ar.getArtistName()).setData(ar.getData().toString()).setLocation(ar.getLocation()).setAvailableSeats(ar.getAvailableSeats().toString()).setSoldSeats(ar.getSoldSeats().toString()).build();
            response.addAllArtisticRepresentation(Collections.singleton(arDTO));
        }

        return response.build();
    }

    public static TSProtobufs.TSResponse createGetAllARFromDateResponse(Iterable<ArtisticRepresentation> ars){
        TSProtobufs.TSResponse.Builder response = TSProtobufs.TSResponse.newBuilder()
                .setType(TSProtobufs.TSResponse.Type.GetAllARFromDate);

        for(ArtisticRepresentation ar : ars){
            TSProtobufs.ArtisticRepresentation arDTO = TSProtobufs.ArtisticRepresentation.newBuilder().setId(ar.getId().toString()).setArtistName(ar.getArtistName()).setData(ar.getData().toString()).setLocation(ar.getLocation()).setAvailableSeats(ar.getAvailableSeats().toString()).setSoldSeats(ar.getSoldSeats().toString()).build();
            response.addAllArtisticRepresentation(Collections.singleton(arDTO));
        }

        return response.build();
    }

    public static TSProtobufs.TSResponse createFindSellerByUsernameResponse(Seller seller){
        TSProtobufs.Seller sellerDTO = TSProtobufs.Seller.newBuilder().setId(seller.getId().toString()).setFirstName(seller.getFirstName()).setLastName(seller.getLastName()).setUsername(seller.getUsername()).setPassword(seller.getPassword()).build();

        TSProtobufs.TSResponse response = TSProtobufs.TSResponse.newBuilder()
                .setType(TSProtobufs.TSResponse.Type.FindSellerByUsername)
                .setSeller(sellerDTO).build();
        return response;
    }

    public static String getError(TSProtobufs.TSResponse response){
        String errorMessage=response.getError();
        return errorMessage;
    }

    public static Seller getSeller(TSProtobufs.TSResponse response){
        Seller seller = new Seller();
        seller.setId(Long.parseLong(response.getSeller().getId()));
        seller.setFirstName(response.getSeller().getFirstName());
        seller.setLastName(response.getSeller().getLastName());
        seller.setUsername(response.getSeller().getUsername());
        seller.setPassword(response.getSeller().getPassword());
        return seller;
    }

    public static Buyer getBuyer(TSProtobufs.TSResponse response){
        Buyer buyer = new Buyer(1L,"scd","sdc",2L);
        buyer.setId(Long.parseLong(response.getBuyer().getId()));
        buyer.setIdAR(Long.parseLong(response.getBuyer().getIdAR()));
        buyer.setFirstName(response.getBuyer().getFirstName());
        buyer.setLastName(response.getBuyer().getLastName());
        buyer.setNoTickets(Long.parseLong(response.getBuyer().getNoTickets()));
        return buyer;
    }

    public static ArtisticRepresentation getArtisticRepresentation(TSProtobufs.TSResponse response){
        ArtisticRepresentation ar = new ArtisticRepresentation("1L", LocalDateTime.now(),"sdc",2L, 2L);
        ar.setId(Long.parseLong(response.getArtisticRepresentation(0).getId()));
        ar.setArtistName(response.getArtisticRepresentation(0).getArtistName());
        //ar.setData(LocalDateTime.parse(response.getArtisticRepresentation(0).getData()));
        ar.setData(getDateFromString(response.getArtisticRepresentation(0).getData()));
        ar.setLocation(response.getArtisticRepresentation(0).getLocation());
        ar.setAvailableSeats(Long.parseLong(response.getArtisticRepresentation(0).getAvailableSeats()));
        ar.setSoldSeats(Long.parseLong(response.getArtisticRepresentation(0).getSoldSeats()));
        return ar;
    }

    public static Iterable<ArtisticRepresentation> getArtisticRepresentations(TSProtobufs.TSResponse response){
        List<ArtisticRepresentation> ars = new ArrayList<>();
        for(int i=0;i<response.getArtisticRepresentationCount();i++){
            ArtisticRepresentation ar = new ArtisticRepresentation("1L", LocalDateTime.now(),"sdc",2L, 2L);
            ar.setId(Long.parseLong(response.getArtisticRepresentation(i).getId()));
            ar.setArtistName(response.getArtisticRepresentation(i).getArtistName());
            //ar.setData(LocalDateTime.parse(response.getArtisticRepresentation(i).getData()));
            ar.setData(getDateFromString(response.getArtisticRepresentation(i).getData()));
            ar.setLocation(response.getArtisticRepresentation(i).getLocation());
            ar.setAvailableSeats(Long.parseLong(response.getArtisticRepresentation(i).getAvailableSeats()));
            ar.setSoldSeats(Long.parseLong(response.getArtisticRepresentation(i).getSoldSeats()));

            ars.add(ar);
        }
        return ars;
    }

    public static LocalDateTime getDateFromString(String raw){
        String[] datatime = raw.split(" ");
        String[] data = datatime[0].split("/");
        String[] time = datatime[1].split(":");
        LocalDateTime result = LocalDateTime.of(Integer.parseInt(data[2]), Integer.parseInt(data[1]), Integer.parseInt(data[0]),Integer.parseInt(time[0]), Integer.parseInt(time[1]));
        return result;
    }

    public static Seller getSeller(TSProtobufs.TSRequest request){
        Seller seller = new Seller();
        seller.setId(Long.parseLong(request.getSeller().getId()));
        seller.setFirstName(request.getSeller().getFirstName());
        seller.setLastName(request.getSeller().getLastName());
        seller.setUsername(request.getSeller().getUsername());
        seller.setPassword(request.getSeller().getPassword());
        return seller;
    }

    public static Buyer getBuyer(TSProtobufs.TSRequest request){
        Buyer buyer = new Buyer(1L,"scd","sdc",2L);
        buyer.setId(Long.parseLong(request.getBuyer().getId()));
        buyer.setIdAR(Long.parseLong(request.getBuyer().getIdAR()));
        buyer.setFirstName(request.getBuyer().getFirstName());
        buyer.setLastName(request.getBuyer().getLastName());
        buyer.setNoTickets(Long.parseLong(request.getBuyer().getNoTickets()));
        return buyer;
    }

    public static ArtisticRepresentation getArtisticRepresentation(TSProtobufs.TSRequest request){
        ArtisticRepresentation ar = new ArtisticRepresentation("1L", LocalDateTime.now(),"sdc",2L, 2L);
        ar.setId(Long.parseLong(request.getArtisticRepresentation().getId()));
        ar.setArtistName(request.getArtisticRepresentation().getArtistName());
        ar.setData(LocalDateTime.parse(request.getArtisticRepresentation().getData()));
        ar.setLocation(request.getArtisticRepresentation().getLocation());
        ar.setAvailableSeats(Long.parseLong(request.getArtisticRepresentation().getAvailableSeats()));
        ar.setSoldSeats(Long.parseLong(request.getArtisticRepresentation().getSoldSeats()));
        return ar;
    }




}
